src/holofuel/pages/Inbox/Inbox.js

```diff
@@ -168,3 +167,12 @@ function ConfirmRejectionModal ({ transaction, handleClose, declineTransaction }
     </div>
   </Modal>
 }
+
+export function RenderNickname ({ agentId }) {
```

Comment:
I feel a little uneasy about this approach, but I'm not sure that means it's wrong. 

Did you try the approach of batch fetching all the nicknames at once in Inbox component?

I can see pros and cons to either way. Let's talk about this.